### PR TITLE
Pruned duplicated dependencies

### PR DESCRIPTION
**Pull request by koushiro**
update some dependencies to pruned duplicated dependencies

- update `comfy-table` v0.5.1 => v6.0.0
- update `strum` v0.23.0  => v0.24.1
- update `h2` v0.3.9 => v0.3.13
- update `file-per-thread-logger` v0.1.4 => v0.1.5
- update `mio` v0.8.0 => v0.8.4
- update `secp256k1` v0.21.2 => v0.24.0

polkadot companion: https://github.com/paritytech/polkadot/pull/5808

### CODE REVIEW

File: primitives/core/src/ecdsa.rs [hunk 1]
```diff
@@ -499,7 +499,7 @@ impl TraitPair for Pair {
 impl Pair {
 	/// Get the seed for this key.
 	pub fn seed(&self) -> Seed {
-		self.secret.serialize_secret()
+		self.secret.secret_bytes()
```
Comment (kianenigma) [inline primitives/core/src/ecdsa.rs:502]: 👀

**Review by kianenigma — APPROVED** [review]

File: primitives/core/src/ecdsa.rs (hunk 1 above)
Comment (koushiro) [reply primitives/core/src/ecdsa.rs:502]: it's a breaking change of secp256k1 api

Comment (ggwpez) [reply primitives/core/src/ecdsa.rs:502]: Just a rename indeed [commit](https://github.com/rust-bitcoin/rust-secp256k1/commit/5c7c76eb7494d3a031301d40d1ccebd0f9b9d7ae).

**Review by ggwpez — APPROVED** [review]

**Comment by bkchr** [comment]
@koushiro please merge master to make the regression benchmark job work.

**Review by bkchr — APPROVED** [review]

**Comment by bkchr** [comment]
bot merge

**Comment by bkchr** [comment]
bot merge

**Comment by bkchr** [comment]
bot merge

**Pull request merged**